compiler/sem/semstmts.nim

```diff
@@ -116,9 +116,10 @@ proc fixNilType(c: PContext; n: PNode) =
 
 # start `discard` check related code
 
-proc implicitlyDiscardable(n: PNode): bool =
-  var n = n
-  while n.kind in skipForDiscardable: n = n.lastSon
+func implicitlyDiscardable(n: PNode): bool =
+  ## true if a node, `n`, is implicitly discardable, meaning it's a control
+  ## flow ending statement in a block (`return`, `break`, `raise`, etc) or a
+  ## routine call to a routine type marked as `{.discardable.}`.
```

Comment:
```suggestion
  ## routine call to a routine marked as `{.discardable.}`.
```
A soft suggestion. Since the result being discardable is not a property of the procedure's type, mentioning its type could be confusing.

Reply:
> I know we mark it on the symbol, but the symbol does multiple duties, one of which is a stand-in for the type of the procedure.

My suggestion was more in the context of the implementation side of things (should've made that clearer, sorry), but I agree that whether a value is automatically discardable makes more sense as a type property.

I do think it's more a property of the *return* type rather than directly one of the *procedure* type. In any case, I don't think it's a problem if the comment stays as it is.